OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -284,19 +285,57 @@ void onMessageActionOccurredOnPreview(@NonNull final OSInAppMessage message, @No
         action.firstClick = message.takeActionAsUnique();
 
         firePublicClickHandler(action);
+        showPrompts(message, action.prompts);
         fireClickAction(action);
+        fireTagsAndOutcomeLogs(action);
     }
 
-    private void firePrompt(OSInAppMessage message, final List<OSInAppMessagePrompt> prompts) {
-        for (OSInAppMessagePrompt prompt : prompts) {
+    private void fireTagsAndOutcomeLogs(final OSInAppMessageAction action) {
```

Comment:
Don't make it specific to tags and outcomes we might have more things added here eventually

Change name from `fireTagsAndOutcomeLogs` to
`logInAppMessagePreviewActions` or
`logIamPreviewActions`?